geowrangler/datasets/ookla.py

```diff
@@ -41,14 +68,208 @@ def list_ookla_files() -> dict:
     return keys
 
 
+# Cell
+
+
+def lookup_ookla_file(filename):
+    """Get OoklaFile for the given filename"""
+    ookla_files = L(list_ookla_files().items())  # use fastcore L as list
+    idx = ookla_files.argfirst(lambda o: o[1] == filename)  # find matching entry
+    return ookla_files[idx][0] if idx is not None else None
+
+
+# Cell
+
+
+def compute_datakey(aoi_bounds, type_, year, return_geometry):
+    data_tuple = (
+        np.array2string(aoi_bounds, precision=6),
+        str(type_),
+        str(year),
+        str(return_geometry),
+    )
+    m = hashlib.md5()
+    for item in data_tuple:
+        m.update(item.encode())
+    data_key = m.hexdigest()
+    return data_key
+
+
+# Cell
+
+
+def write_ookla_metajson(
+    cache_dir, data_key, total_bounds, type_, year, return_geometry
+):
+    cached_metajson_file_path = (
+        os.path.join(cache_dir, f"{data_key}.geojson.metadata.json")
+        if return_geometry
+        else os.path.join(cache_dir, f"{data_key}.csv.metadata.json")
+    )
+    with open(cached_metajson_file_path, "w") as f:
+        f.write(
+            json.dumps(
+                {
+                    "bounds": np.array2string(total_bounds, precision=6),
+                    "type_": type_,
+                    "year": year,
+                    "with_geom": return_geometry,
+                }
+            )
+        )
+
+
+# Cell
+
+
+class OoklaDataManager:
+    """An instance of this class provides convenience functoins for loading and caching Ookla data"""
+
+    def __init__(self, cache_dir=DEFAULT_CACHE_DIR):
+        self.data_cache = {}
+        self.cache_dir = os.path.expanduser(cache_dir)
+        self.processed_cache_dir = os.path.join(self.cache_dir, "ookla/processed/")
+        Path(self.processed_cache_dir).mkdir(parents=True, exist_ok=True)
+
+
+# Cell
+
+
+@patch
+def reinitialize_processed_cache(self: OoklaDataManager):
+    "Reinitialize processed_cache_dir to start over from scratch."
+    shutil.rmtree(self.processed_cache_dir, ignore_errors=True)
+    response = Path(self.processed_cache_dir).mkdir(parents=True, exist_ok=True)
+    logger.info(
+        f"{self.processed_cache_dir} reintialized. All cached processed data in this folder has been deleted."
+    )
+    return response
+
+
+# Cell
+
+
+@patch
+def load_type_year_data(
```

Comment:
I'm not convinced this is the best way to load data for ookla for aoi. Loading entire geopandas dataframe seems wasterful. Ookla has shapefiles along side parquet which we can use the bbox parameter of geopandas to partial read.

For example, downloading aoi 
```
gdf = gpd.read_file("2019-01-01_performance_fixed_tiles.zip", bbox=(4.0000, 116.1600,5.5800, 126.6000))
```

Technically speaking you can also achieve with parquet
```
pd.read_parquet("2019-01-01_performance_fixed_tiles.parquet", filters=[('quadkey', 'in', ['0022133222312322'])])
````

Reply:
Documenting this as a future optimization enhancement for 
[Ookla](https://github.com/thinkingmachines/geowrangler/discussions/203)